### PR TITLE
openssl: fix cmake variables

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **openssl/1.***

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
cc @SpaceIm Do you have on how to fix cmake variables creation with CMakeDeps https://github.com/conan-io/conan-center-index/blob/master/recipes/openssl/1.x.x/conanfile.py#L828 ?

File: recipes/openssl/1.x.x/conanfile.py [hunk 1]
```diff
@@ -834,21 +833,33 @@ def _create_cmake_module_variables(module_file):
...
                 set(OPENSSL_SSL_LIBRARIES ${OpenSSL_SSL_LIBS}
                                           ${OpenSSL_SSL_DEPENDENCIES}
                                           ${OpenSSL_SSL_FRAMEWORKS}
                                           ${OpenSSL_SSL_SYSTEM_LIBS})
+            elseif(DEFINED openssl_OpenSSL_SSL_LIBS_%(config)s)
+                set(OPENSSL_SSL_LIBRARY ${openssl_OpenSSL_SSL_LIBS_%(config)s})
+                set(OPENSSL_SSL_LIBRARIES ${openssl_OpenSSL_SSL_LIBS_%(config)s}
+                                          ${openssl_OpenSSL_SSL_DEPENDENCIES_%(config)s}
+                                          ${openssl_OpenSSL_SSL_FRAMEWORKS_%(config)s}
+                                          ${openssl_OpenSSL_SSL_SYSTEM_LIBS_%(config)s})
```
Comment (ericLemanissier) [inline recipes/openssl/1.x.x/conanfile.py:854]: I feel dirty after writing this, but what other choice do we have ?

**Comment by ericLemanissier** [comment]
blocks:
- https://github.com/conan-io/conan-center-index/pull/12906

**Comment by ericLemanissier** [comment]
Re-ping @SpaceIm what do you think of this (ugly) solution?

File: recipes/openssl/1.x.x/test_package/conanfile.py [hunk 2]
```diff
@@ -22,21 +24,32 @@ def _skip_test(self):
...
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if self.settings.os == "Android":
+            tc.cache_variables["CONAN_LIBCXX"] = ""
+        openssl_version = Version(self.deps_cpp_info["openssl"].version)
+        if openssl_version.major == "1" and openssl_version.minor == "1":
+            tc.cache_variables["OPENSSL_WITH_ZLIB"] = False
+        else:
+            tc.cache_variables["OPENSSL_WITH_ZLIB"] = not self.options["openssl"].no_zlib
```
Comment (uilianries) [inline recipes/openssl/1.x.x/test_package/conanfile.py:41]:
```suggestion
        openssl = self.dependencies["openssl"]
        openssl_version = Version(openssl.ref.version)
        if openssl_version.major == "1" and openssl_version.minor == "1":
            tc.cache_variables["OPENSSL_WITH_ZLIB"] = False
        else:
            tc.cache_variables["OPENSSL_WITH_ZLIB"] = not openssl.options.no_zlib
```

docs.conan.io/en/latest/reference/conanfile/dependencies.html

File: recipes/openssl/1.x.x/test_package/conanfile.py [hunk 3]
```diff
@@ -22,21 +24,32 @@ def _skip_test(self):
...
-                cmake.definitions["OPENSSL_WITH_ZLIB"] = not self.options["openssl"].no_zlib
             cmake.configure()
             cmake.build()
 
     def test(self):
-        if not self._skip_test and not cross_building(self):
-            bin_path = os.path.join("bin", "digest")
+        if not self._skip_test and can_run(self):
+            bin_path = os.path.join(self.cpp.build.bindirs[0], "digest")
             self.run(bin_path, run_environment=True)
```
Comment (uilianries) [inline recipes/openssl/1.x.x/test_package/conanfile.py:54]:
```suggestion
            self.run(bin_path, env="conanrun")
```

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

**Comment by jcar87** [comment]
Hi there! 
I've landed on this PR from this issue report: https://github.com/conan-io/conan/issues/12222 

Let's please make sure PR description provide a motivation for the changes - the title mentions "fixing" the variables but no context as to what was broken in the first place, so it's made it a bit harder for me to understand what needs fixing in CMakeDeps :(

**Comment by ericLemanissier** [comment]
Hi, my message https://github.com/conan-io/conan-center-index/pull/13228#issuecomment-1263248392 was supposed to give the context. I'll take care to put it in the PR description next time, if your feel it makes a difference.

**Comment by jcar87** [comment]
> Hi, my message [#13228 (comment)](https://github.com/conan-io/conan-center-index/pull/13228#issuecomment-1263248392) was supposed to give the context. I'll take care to put it in the PR description next time, if your feel it makes a difference.

That would be great, thanks! I feel like that's the PR description is the best place for this, and it _is_ part of the current PR template. Otherwise one has to scroll down the list of messages to work it out.

**Pull request merged**